pkg/executor/util.go

```diff
@@ -68,6 +68,13 @@ func labelsForCluster(
 		ClusterAppLabel:        cr.Spec.AppID,
 		ClusterAppCatalogLabel: *(cr.Spec.AppCatalog),
 	}
+	appCR, err := catalog.GetApp(cr)
+	if err == nil {
+		// Apply app labels to cluster
+		for name, val := range appCR.Labels {
+			result[name] = val
+		}
+	}
```

Comment:
Would we want to put the kdapp's labels on the cluster components (which is what this does) or would we rather put them on the kdcluster itself? Or both?

Reply:
good point, I don't feel strongly one way or the other. I just used the convenience fun that puts crName, appid etc